plugins/include/open62541/plugin/create_certificate.h

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2021 (c) Christian von Arnim, ISW University of Stuttgart (for VDW and umati)
+ */
+
+#ifndef CREATE_CERTIFICATE_H_
+#define CREATE_CERTIFICATE_H_
+
+#include <open62541/plugin/log.h>
+#include <open62541/types.h>
+
+#ifdef UA_ENABLE_ENCRYPTION
+enum UA_CertificateFormat {
+    UA_CERTIFICATE_FORMAT_DER,
+    UA_CERTIFICATE_FORMAT_PEM
+};
+
+/**
+ * (Experimental) Create a self-signed certificate
+ *
+ * It is recommended to store the generated certificate in a non volatile memory and
+ * reuse it, so the application can be recognized across several executions.
+ *
+ * \param subject Elements for the subject,
+ *                  e.g. ["C=DE", "O=SampleOrganization", "CN=Open62541Server@localhost"]
+ * \param subjectAltName Elements for SubjectAltName,
+ *                  e.g. ["DNS:localhost", "URI:urn:open62541.server.application"]
+ */
+UA_StatusCode UA_EXPORT
+UA_CreateCertificate(const UA_Logger *logger,
```

Comment:
Please use pointers instead of array syntax to be consistent with the rest of the SDK.

Is it possible to generalize this function for different key lengths?
That would be great so we don't have to change the API later when a different key length is requested.

Reply:
Parameter is added.
In case a `0` is provided, the largest possible key is generated (4096Bits in the current OPC UA Specification).